src/software/telemetry/src/structures.rs

```diff
@@ -30,6 +32,27 @@ pub enum AlarmPriority {
     Low,
 }
 
+impl PartialOrd for AlarmPriority {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for AlarmPriority {
+    fn cmp(&self, other: &Self) -> Ordering {
```

Comment:
Wouldn't that be more readable to assign an int to each priority and call `.cmp()` to handle the ordering?

Reply:
You mean having :
```
pub enum AlarmPriority {
    High = 1,
    Medium = 2,
    Low = 3,
}
```

And compare the numbers? If so yes it could work but IIRC you would then be able to do `4 as AlarmPriority` and that's undefined behaviour.

Or maybe
```
impl AlarmPriority {
  fn as_int(self) -> u8 {
     match self {
       Self::High => 1,
        ....
     }
  }
}
```
And then I guess we could compare

Reply:
I didn't overthought this ^^ I had something like this in mind (untested):

```rust
impl Ord for AlarmPriority {
    fn cmp(&self, other: &Self) -> Ordering {
        let priorityToInt = |priority: AlarmPriority| match priority {
            AlarmPriority::High => 3,
            AlarmPriority::Medium => 3,
            AlarmPriority::Low => 3,
        };
        priorityToInt(self).cmp(priorityToInt(other))
    }
}
```

But if you think this should not be done locally because we have other use cases for this "AlarmPriority → Int" mapping, do what's best!

Reply:
that's a nice idea, for now I don't think we have such use case so I'll just go with it